lib/Dialect/RTG/Transforms/EmitRTGISAAssemblyPass.cpp

```diff
@@ -0,0 +1,249 @@
+//===- EmitRTGISAAssemblyPass.cpp - RTG Assembly Emitter ------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This is the main ISA Assembly emitter implementation for the RTG dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Dialect/RTG/IR/RTGISAAssemblyOpInterfaces.h"
+#include "circt/Dialect/RTG/IR/RTGOps.h"
+#include "circt/Dialect/RTG/Transforms/RTGPasses.h"
+#include "circt/Support/Path.h"
+#include "mlir/IR/Threading.h"
+#include "mlir/Support/FileUtilities.h"
+#include "llvm/ADT/SmallString.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/ADT/StringSet.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/Path.h"
+#include "llvm/Support/ToolOutputFile.h"
+#include "llvm/Support/raw_ostream.h"
+#include <fstream>
+
+namespace circt {
+namespace rtg {
+#define GEN_PASS_DEF_EMITRTGISAASSEMBLYPASS
+#include "circt/Dialect/RTG/Transforms/RTGPasses.h.inc"
+} // namespace rtg
+} // namespace circt
+
+using namespace circt;
+using namespace rtg;
+
+#define DEBUG_TYPE "emit-rtg-isa-assembly"
+
+namespace {
+
+class Emitter {
+public:
+  Emitter(llvm::raw_ostream &os,
+          const llvm::StringSet<llvm::MallocAllocator> &unsupportedInstr)
+      : os(os), unsupportedInstr(unsupportedInstr) {}
+
+  LogicalResult emitInstruction(InstructionOpInterface instr) {
+    os << llvm::indent(4);
+    bool useBinary = unsupportedInstr.contains(instr->getName().getStringRef());
```

Comment:
The name internally is stored as an Attribute.  A set over StringAttr would likely be more efficient than a StringSet.

Reply:
Which obviously requires constructing StringAttrs for each unsupoorted instruction at initialization time, but that's a one time cost.